### PR TITLE
Add  IssueFilter for detailed issue query & FixedVersion field to issue

### PR DESCRIPTION
**Pull request by getgot78**
Hi, I'd appreciate it if you could merge the following :
* add IssueFilter struct & method for more detailed query in issue collection.
* add FixedVersion field for version information.

### CODE REVIEW

File: issue.go [hunk 1]
```diff
@@ -32,13 +33,21 @@ type Issue struct {
...
 	Author       *IdName        `json:"author"`
+	FixedVersion *IdName        `json:"fixed_version"`
 	AssignedTo   *IdName        `json:"assigned_to"`
 	Notes        string         `json:"notes"`
 	StatusDate   string         `json:"status_date"`
 	CreatedOn    string         `json:"created_on"`
 	UpdatedOn    string         `json:"updated_on"`
 	CustomFields []*CustomField `json:"custom_fields,omitempty"`
 }
+type IssueFilter struct {
```
Comment (mattn) [inline issue.go]: add empty line

File: issue.go [hunk 2]
```diff
@@ -121,6 +130,29 @@ func (c *client) IssuesByQuery(query_id int) ([]Issue, error) {
 	}
 	return r.Issues, nil
 }
+func (c *client) IssuesByFilter(f IssueFilter) ([]Issue, error) {
```
Comment (mattn) [inline issue.go]: ditto

File: issue.go [hunk 3]
```diff
@@ -32,13 +33,21 @@ type Issue struct {
...
+	FixedVersion *IdName        `json:"fixed_version"`
 	AssignedTo   *IdName        `json:"assigned_to"`
 	Notes        string         `json:"notes"`
 	StatusDate   string         `json:"status_date"`
 	CreatedOn    string         `json:"created_on"`
 	UpdatedOn    string         `json:"updated_on"`
 	CustomFields []*CustomField `json:"custom_fields,omitempty"`
 }
+type IssueFilter struct {
+	ProjectId    string
```
Comment (mattn) [inline issue.go]: add json tag for all

File: issue.go [hunk 4]
```diff
@@ -121,6 +130,29 @@ func (c *client) IssuesByQuery(query_id int) ([]Issue, error) {
 	}
 	return r.Issues, nil
 }
+func (c *client) IssuesByFilter(f IssueFilter) ([]Issue, error) {
+	res, err := http.Get(c.endpoint + "/issues.json?key=" + c.apikey + c.getPaginationClause() + getIssueFilterClause(f))
```
Comment (mattn) [inline issue.go]: getIssueFilterClause(&f)

File: issue.go [hunk 5]
```diff
@@ -245,3 +277,24 @@ func (c *client) DeleteIssue(id int) error {
 func (issue *Issue) GetTitle() string {
 	return issue.Tracker.Name + " #" + strconv.Itoa(issue.Id) + ": " + issue.Subject
 }
+
+func getIssueFilterClause(filter IssueFilter) string {
```
Comment (mattn) [inline issue.go]: func getIssueFilterClause(filter *IssueFilter) string

File: issue.go [hunk 6]
```diff
@@ -245,3 +277,24 @@ func (c *client) DeleteIssue(id int) error {
...
+	clause := ""
+	if filter.ProjectId != "" {
+		clause = clause + fmt.Sprintf("&project_id%v", filter.ProjectId)
```
Comment (mattn) [inline issue.go]: you forgot `=`?

File: issue.go (hunk 3 above)
Comment (getgot78) [inline issue.go]: I guess these tags are for json response, but this sturct 'IssueFilter' is not for json response, just use in argument of method.Would you need these?

File: issue.go (hunk 6 above)
Comment (getgot78) [inline issue.go]: That is as I intended.
According to redmaine Rest-Api wiki, you could use ">" or "<" with these queries too, 
url : http://www.redmine.org/projects/redmine/wiki/Rest_Issues
it's simple solution.

File: issue.go (hunk 1 above)
Comment (getgot78) [inline issue.go]: ok

File: issue.go (hunk 2 above)
Comment (getgot78) [inline issue.go]: ok

File: issue.go (hunk 5 above)
Comment (getgot78) [inline issue.go]: I got it too.

File: issue.go (hunk 3 above)
Comment (mattn) [inline issue.go]: Sorry, This is my mistake. Ignore this.

File: issue.go [hunk 7]
```diff
@@ -278,22 +280,22 @@ func (issue *Issue) GetTitle() string {
 	return issue.Tracker.Name + " #" + strconv.Itoa(issue.Id) + ": " + issue.Subject
 }
 
-func getIssueFilterClause(filter IssueFilter) string {
+func getIssueFilterClause(filter *IssueFilter) string {
 	clause := ""
```
Comment (mattn) [inline issue.go]: if filter == nil {
    return ""
}

**Comment by mattn** [comment]
Thanks.

**Comment by getgot78** [comment]
Likewise!

**Pull request merged**